marimo/_ast/app.py

```diff
@@ -777,6 +783,17 @@ async def embed(self) -> AppEmbedResult:
         notebook that in turn embeds another notebook, and marimo will do the
         right thing.
 
+         Notes on parent->child exposure:
+        - If running in a notebook, `expose` installs a **read-only** namespace that
+          is called `parent` by default inside the child kernel. The child must make a
+          **static reference** to that name (e.g., `parent` or `from parent import ...`)
+          for the analyzer to mark dependencies and enable targeted re-runs.
+        - If `namespace=None`, exposed names are injected flat into child globals. This
+          is advanced, may be shadowed by child definitions, and should be avoided
+          unless that behavior is desired.
+        - In script mode, exposure is a one-time snapshot without reactivity. The namespace
+          is still read-only to prevent accidental mutation.
```

Comment:
I think `namespace` scoping and `read-only` add potentially unneeded complexity. This logic could be downstreamed to mowidgets, and once a stable api we can upstream it